python/lsst/pipe/tasks/transformMeasurement.py

```diff
@@ -152,38 +138,85 @@ class RunTransformTask(pipeBase.CmdLineTask):
-        dataRef.put(outputCat, "transformedSrc")
+        dataRef.put(outputCat, "transformed" + self.config.sourceType.capitalize())
         return outputCat
+
+
+def makeTransformCmdLineTask(sourceType, calexpType, measConfigType, outputType, wasForced):
+    """!Generate and return a specialization of RunTransformTaskBase.
+
+    RunTransformTaskBase provides a means of transforming arbitrary source types (src, forced_src,
+    deepCoadd_src, ...). However, a specialization is required for each type we wish to transform. Here, we
+    generate a specialization for the source type required.
+
+    @param[in] sourceType      Name of source type to transform (e.g src, forced_src, etc).
+    @param[in] calexpType      Name of exposure with calibration data (e.g. calexp, etc)
+    @param[in] measConfigType  Name of measurement task configuration (e.g. processCCd_config, etc)
+    @param[in] outputType      Name of output type to write (e.g. transformedSrc, etc)
+    @param[in] wasForced       Boolean indicating if source measurements were forced.
+
+    @returns A tuple of (Config, CmdLineTask) suitable for transforming sources of the given type.
+    """
+    sourceTypeField = pexConfig.Field(dtype=str, default=sourceType,
+                                      doc="Dataset type of source to transform")
+    calexpTypeField = pexConfig.Field(dtype=str, default=calexpType,
+                                      doc="Dataset type of calibration exposure")
+    measConfigField = pexConfig.Field(dtype=str, default="processCcd_config",
+                                      doc="Dataset type of measurement operation configuration")
+    outputTypeField = pexConfig.Field(dtype=str, default=outputType,
+                                      doc="Dataset type of output sources")
+    forcedField = pexConfig.Field(dtype=bool, default=wasForced, doc="Measurement operation was forced")
+
+    configClass = type("%sTransformConfig" % sourceType.capitalize(),
+                       (RunTransformConfigBase, pexConfig.Config),
+                       {"sourceType": sourceTypeField, "calexpType": calexpTypeField,
+                        "outputType": outputTypeField, "measConfig": measConfigField,
+                        "wasForced": forcedField})
+    taskClass = type("%sTransformTask" % (sourceType.capitalize(),),
+                     (RunTransformTaskBase, pipeBase.CmdLineTask),
+                     {"wasForced": wasForced, "ConfigClass": configClass,
+                      "_DefaultName": "transform%sMeasurement" % (sourceType.capitalize(),)})
+    return configClass, taskClass
+
+
+# Explicitly instantiate command line transformation types for common source types. Camera-specific
+# transformation tasks can be defined following this pattern in the appropriate obs_ package.
+SrcTransformConfig, SrcTransformTask = makeTransformCmdLineTask(
+                                                "src", "calexp", "processCcd_config", "transformedSrc", False)
```

Comment:
Instead of "transformedSrc", let's go with "transformed_src", so we're just adding a prefix and not changing any capitalization (we don't at present have any naming conventions for butler datasets, so either is legal).  @RobertLuptonTheGood has analysis code that can operate generically on different types of data by just using different prefixes, and while we may not use that in the long run I think not requiring extra camelcase smarts to machine-generate related names from each other is good practice in general.

In fact, I think it'd be entirely appropriate for all of these tasks to essentially hard code that their output dataset is "transformed_" prepended to the input name; allowing flexibility doesn't seem to have any advantages.